src/leidenalg/GraphHelper.cpp

```diff
@@ -284,12 +288,12 @@ int Graph::has_self_loops()
   return has_self_loops;
 }
 
-size_t Graph::possible_edges()
+double Graph::possible_edges()
 {
   return this->possible_edges(this->vcount());
 }
 
-size_t Graph::possible_edges(size_t n)
+double Graph::possible_edges(double n)
 {
   size_t possible_edges = n*(n-1);
```

Comment:
this should be `double`, but the formula is not correct for non-integers, right?

Reply:
also, below, shouldn't the /2 be after the +=n ?

Reply:
The interpretation becomes a bit fuzzy with non-integer sizes indeed, but I wouldn't say it is "incorrect", just "unclear". 

You can argue about the `/= 2`, depending on the interpretation of a self-loop. If you count each pair of nodes $i \neq j$ once, you get $\frac{n(n-1)}{2}$ possible edges. In addition, there are then $n$ self-loops possible (not $n / 2$). Hence, from that point of view, the total number of possible edges is correct.

Reply:
Yes, I wasn't quite sure how to interpret it with non-integer values, but couldn't figure out how to 'fix' it, so 'unclear' is indeed a better name haha. You're right for the other comments, I didn't think about it long enough!